src/models/transformer.h

```diff
@@ -285,7 +285,8 @@ class Transformer : public EncoderOrDecoderBase {
       kh = cache_[prefix + "_keys"];                                                   // then return cached tensor
     }
     else {
-      auto Wk = graph_->param(prefix + "_Wk", {dimModel, dimModel}, inits::glorotUniform());
+      int dimKeys =  keys->shape()[-1];
+      auto Wk = graph_->param(prefix + "_Wk", {dimKeys, dimModel}, inits::glorotUniform());
```

Comment:
The weight attention transformer matrices were being initialized with shape [dimModel x dimModel]. If concatenation is chosen and we are only using source factors, the embedding size of the encoder and decoder will be different, so in the enc-dec attention layer of the decoder, we need to initialize the weight matrices of keys and values with whatever dimension is outputted from the encoder.